source/docs/advanced/theme/analytics.md

```diff
@@ -0,0 +1,237 @@
+
+Additionally, you may wonder what name and properties you should give to your events. It depends on the integrations you are using and if you want to create a new event or use Semantic ones (supported events within Segment.io). To learn more about this, please refere to the [Semantic Events documentation](https://segment.com/docs/spec/semantic/).
+
+### Track an event as a React Component
+
+If you don't have an actual callback where to put the `trackEvent` (like `onClick`), you can use the `withTrackOnMount` enhancer that will let you call the `trackEvent` using [React lifecycle]((http://projects.wojtekmaj.pl/react-lifecycle-methods-diagram/)).
+
+```jsx
+import withTrackOnMount from "theme/modules/Analytics/withTrackOnMount";
+
+withTrackOnMount({
+  // event name
+  event,
+  // compute events props
+  mapPropsToProperties: props => ({ ...properties })
+  // is the component ready to be tracked?
+  isResolvedFromProps: props => true || false,
+  // should we listen the mount during the lifecycle?
+  shouldMountEvent: props => true || false,
+  // should we listen the update during the lifecycle?
+  shouldUpdateEvent: (prevProps, nextProps) => true || false,
+})(BaseComponent);
+```
+
+If you prefer, you can even use the `TrackOnMount` component that works with render props.
+
+```jsx
+import React, { Fragment } from "react";
+import { TrackOnMount } from "theme/modules/Analytics/withTrackOnMount";
+
+const Component = (props) => {
+  return <Fragment>
+    <TrackOnMount
+        shouldMountEvent={props => true || false}
+        shouldUpdateEvent={(prevProps, nextProps) => true || false}
+        event={event}
+        properties={eventProperties}
+        isResolvedFromProps={true || false}
+    />
+    {/* ... your actual component code here */}
+  </Fragment>;
+}
+```
+
+### Track page
+
+In trackings scripts, there is often a distinction between the `page` and the `event` even though a `page` event is only a subset of the `event`s.
+
+To make this distinction clear, we provide APIs in the same spirit of `withTrackOnMount` and `TrackOnMount` but for page events.
+
+* `withTrackPage` is an enhancer that lets you track a page each time a component is rendered and when the location has changed
+```jsx
+import withTrackPage from "theme/modules/Analytics/withTrackPage";
+
+withTrackPage("My Page")(BaseComponent);
+```
+* `TrackPage` is a component using the Render Props paradigm
+```jsx
+import React, { Fragment } from "react";
+import { TrackPage } from "theme/modules/Analytics/withTrackPage";
+
+const Component = (props) => {
+  return <Fragment>
+    <TrackPage title="My Page" />
+    {/* ... your actual component code here */}
+  </Fragment>;
+}
+```
+
+<blockquote class="note">
+Note that we didn't talk about a `trackPage` method here. This is because a `Page` is tighlty coupled to a React Component. This is why you shouldn't need to use `trackPage` directly.
```

Comment:
```suggestion
Note that we didn't talk about a `trackPage` method here. This is because a `Page` is tightly coupled to a React Component. This is why you shouldn't need to use `trackPage` directly.
```